File: backend/src/main/java/com/soccerdreamfermana/controller/DonationController.java
package com.soccerdreamfermana.controller;

import com.soccerdreamfermana.dto.donation.CapturePaymentRequest;
import com.soccerdreamfermana.dto.donation.CapturePaymentResponse;
import com.soccerdreamfermana.dto.donation.CreatePaymentRequest;
import com.soccerdreamfermana.dto.donation.CreatePaymentResponse;
import com.soccerdreamfermana.dto.donation.DonationRequest;
import com.soccerdreamfermana.dto.donation.DonationResponse;
import com.soccerdreamfermana.service.DonationService;
import jakarta.validation.Valid;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/donations")
@RequiredArgsConstructor
public class DonationController {

 private final DonationService service;

 @PostMapping("/create-payment")
 public ResponseEntity<CreatePaymentResponse> createPayment(@Valid @RequestBody CreatePaymentRequest request) {
  return ResponseEntity.ok(service.createPayment(request));
 }

 @PostMapping("/capture-payment")
 public ResponseEntity<CapturePaymentResponse> capturePayment(@Valid @RequestBody CapturePaymentRequest request) {
  return ResponseEntity.ok(service.capturePayment(request.orderId()));
 }

 @PostMapping
 public DonationResponse create(@Valid @RequestBody DonationRequest request) {
  return service.create(request);
 }

 @GetMapping
 public List<DonationResponse> list() {
  return service.list();
 }

 @GetMapping("/{id}")
 public DonationResponse getById(@PathVariable Long id) {
  return service.getById(id);
 }

 @DeleteMapping("/{id}")
 public ResponseEntity<Void> delete(@PathVariable Long id) {
  service.delete(id);
  return ResponseEntity.noContent().build();
 }
}

File: backend/Controllers/DonationController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using SoccerDreamFermana.Backend.Dtos;
using SoccerDreamFermana.Backend.Services;

namespace SoccerDreamFermana.Backend.Controllers;

[ApiController]
[Route("api/donations")]
public sealed class DonationController(DonationService donationService) : ControllerBase
{
    [HttpPost("create-payment")]
    [AllowAnonymous]
    public Task<CreatePaymentResponse> CreatePayment(CreatePaymentRequest request, CancellationToken cancellationToken)
    {
        return donationService.CreatePaymentAsync(request, cancellationToken);
    }

    [HttpPost("capture-payment")]
    [AllowAnonymous]
    public Task<CapturePaymentResponse> CapturePayment(CapturePaymentRequest request, CancellationToken cancellationToken)
    {
        return donationService.CapturePaymentAsync(request.OrderId, cancellationToken);
    }

    [HttpPost]
    [AllowAnonymous]
    public Task<DonationResponse> Create(DonationRequest request, CancellationToken cancellationToken)
    {
        return donationService.CreateAsync(request, cancellationToken);
    }

    [HttpGet]
    [Authorize(Roles = "ADMIN")]
    public Task<IReadOnlyList<DonationResponse>> List(CancellationToken cancellationToken)
    {
        return donationService.ListAsync(cancellationToken);
    }

    [HttpGet("{id:long}")]
    [Authorize(Roles = "ADMIN")]
    public Task<DonationResponse> GetById(long id, CancellationToken cancellationToken)
    {
        return donationService.GetByIdAsync(id, cancellationToken);
    }

    [HttpDelete("{id:long}")]
    [Authorize(Roles = "ADMIN")]
    public async Task<IActionResult> Delete(long id, CancellationToken cancellationToken)
    {
        await donationService.DeleteAsync(id, cancellationToken);
        return NoContent();
    }
}
